### ISSUE TITLE
User Permissions not show correctly when ManageSecurityWithUserGroups = true

### ISSUE THREAD

**Issue by mcl-sz** [umbraco/Umbraco.Forms.Issues#1102]
When ManageSecurityWithUserGroups is set to True, the permissions under "User permissions" are not shown correctly.
The toggles seem to indicate that a user has rights to a/all form, while this is not the case (and should not be the case).

Reproduction
------------

- Change ManageSecurityWithUserGroups to True in UmbracoForms.config
- Create a new user (in this example my user has id=2)
- Create 2 forms
- Under "User permissons" and "Form Security" the toggle is set to true for both forms but the user doesn't have access (which is right).
- Change the toggle for the first form and save.
- Create a new third form.
- Check the permissions under "User permissons" => "Form Security". The toggle for the new form is turned on but the user has no access to the form.

The UFUserFormSecurity table has 2 records where 1 record has HasAccess = 1
![afbeelding](https://github.com/umbraco/Umbraco.Forms.Issues/assets/18614643/ddf66427-2957-47d2-9a55-7cd7e341e254)

### Expected result

I would expect that only the forms that have a record in UFUserFormSecurity where HasAccess is set to 1 have toggled on, just like in "Group permissions". So all the toggles should the turned off initally.

It would be even more nice if the UFUserFormSecurity and UFUserGroupFormSecurity only contained records with HasAccess = 1.
This way it's easier to set the permissions programaticly. Why is it necessary to create records with HasAccess = 0?

### Actual result

The permissions toggles are on, even if the user has no access to the form.

---
_This item has been added to our backlog AB#33964_

**Comment by mcl-sz** [comment]
Hi @AndyButland

I can probably explain it more simply than I did before.

User "Felix Form" is not a member of any groups in this example. There is 1 form called "My Form 1"

Felix is added to "User Permissions".

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/18614643/1632a38e-e6a7-4372-9017-b4370a6b7299)

The toggle for "My Form 1" is now on (it would make more sense if it were off because Felix does not have acces to this form yet)

![afbeelding](https://github.com/umbraco/Umbraco.Forms.Issues/assets/18614643/6cf4fa1e-6919-4d59-b9f8-7c38b43da9bd)

Click Save.

In the "Forms" section, create a second form ("My Form 2").

Return to the "Users" section and select Felix under "User Permissions"

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/18614643/ac9c1a09-4d69-4b2e-83cc-0c0e794cfa6e)

The interface now suggests that Felix has access to "My Form 2", but that is not the case (which makes sense). Only when I click Save again will Felix also have access to "My Form 2".  So the toggle should be off for "My Form 2".


It is now no longer possible to find out from the interface which forms Felix now has acces to because the toggles are on, even if there is no record in the UFUserFormSecurity table.

**Comment by AndyButland** [comment]
OK, thanks for detailing this some more.  I think there are two scenarios.

1. When you are creating a new user security record, the interface indicates that the user has access to the form, when they don't - yet.  I feel this part is OK, as you haven't created the user security record yet.  You haven't pressed Save, and when you do, the record will be created and the user will have either access or not to a form depending on the toggle setting.

So really here all we are doing is pre-selecting the the toggle - and you can choose if you want this pre-select to be on or off based on the setting `DefaultUserAccessToNewForms`.

2. The second part does look like an issue.  Here you have a user security record, and since it was created, a new form has been added.  Now we do have a problem in that you are looking at an existing record, it's showing the user has access to a form, when in fact they don't.

I think we fix that using this same setting.  When rendering a list of forms, for all those that don't exist in the user security record, we show or hide them based on the value of this setting.  I'll look at doing that for a subsequent release.
